### PR TITLE
Using compiled queries

### PR DESCRIPTION
**Pull request by andrewconner**
@eishay @stkem 

All repos now have a `getCompiled` (and ExternalId repos have `getByExtIdCompiled`).

On my machine, on a fresh JVM, 1000 iterations of the old `get` takes 1.2 seconds. Same iterations of the new cached `get` takes 60ms. On a warm JVM, it's 680ms vs 9ms.

Take note of the compiled syntax. Not sure why they use the "two-variable" style in the docs, but this looks clear to me. Also, note that `take` and `drop` are not possible with Slick 2.0 Compiled queries. 2.1 will support this.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserRepo.scala [hunk 1]
```diff
@@ -168,7 +168,7 @@ class UserRepoImpl @Inject() (
 
   override def get(id: Id[User])(implicit session: RSession): User = {
     idCache.getOrElse(UserIdKey(id)) {
-      (for (f <- rows if f.id is id) yield f).firstOption.getOrElse(throw NotFoundException(id))
+      getCompiled(id).first
```
Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserRepo.scala]: does this throw the same exception? Might be problem if it's handled further up and it's no the same.

**Comment by stkem** [comment]
nice!

**Comment by 2is10** [comment]
:cow2: :bell:

**Comment by eishay** [comment]
lgtm

**Pull request merged**